crates/ruff_linter/src/rules/pylint/rules/too_many_branches.rs

```diff
@@ -176,10 +176,11 @@ fn num_branches(stmts: &[Stmt]) -> usize {
                         .sum::<usize>()
             }
             Stmt::Match(ast::StmtMatch { cases, .. }) => {
-                1 + cases
-                    .iter()
-                    .map(|case| num_branches(&case.body))
-                    .sum::<usize>()
+                1 + cases.len()
+                    + cases
+                        .iter()
+                        .map(|case| num_branches(&case.body))
+                        .sum::<usize>()
```

Comment:
I don't think we should count the `match` statement itself as a branch because it's the case blocks which does the branching.


```suggestion
                cases.len()
                    + cases
                        .iter()
                        .map(|case| num_branches(&case.body))
                        .sum::<usize>()
```

Reply:
Makes sense, done.